glad/generator/rust/templates/impl.rs

```diff
@@ -54,20 +65,31 @@ pub mod enumerations {
 }
 
 pub mod functions {
-    #![allow(non_snake_case, unused_variables, dead_code)]
+    #![allow(unused_variables, dead_code)]
 
     use std;
     use std::mem;
-    use super::storage;
+    {{ 'use super::FnPtr;' if options.mx else 'use super::storage;' }}
     use super::types::*;
 
-    {% for command in feature_set.commands %}
-    {{ template_utils.protect(command) }} #[inline] pub unsafe fn {{ command.name|no_prefix }}({{ command|params }}) -> {{ command.proto.ret|type }} { mem::transmute::<_, extern "system" fn({{ command|params('types') }}) -> {{ command.proto.ret|type }}>(storage::{{ command.name|no_prefix }}.ptr)({{ command|params('names') }}) }
-    {% endfor %}
+    {{ 'pub struct Gl {' if options.mx }}
+    {% if options.mx %}
+        {% for command in feature_set.commands %}
+        {{ template_utils.protect(command) }} pub(super) _{{ command.name|no_prefix }}: FnPtr,
+        {% endfor %}
+    }
+
+    impl Gl {
+    {% endif %}
+        {% for command in feature_set.commands %}
+        {{ template_utils.protect(command) }} #[inline] pub unsafe fn {{ command.name|no_prefix }}({{ '&self, ' if options.mx }}{{ command|params }}) -> {{ command.proto.ret|type }} { mem::transmute::<_, extern "system" fn({{ command|params('types') }}) -> {{ command.proto.ret|type }}>({{ 'self._' if options.mx else 'storage::' }}{{ command.name|no_prefix }}.ptr)({{ command|params('names') }}) }
+        {% endfor %}
+    {{ '}' if options.mx }}
```

Comment:
```rust
    {% if options.mx %}
    pub struct Gl {
        {% for command in feature_set.commands %}
        {{ template_utils.protect(command) }} pub(super) _{{ command.name|no_prefix }}: FnPtr,
        {% endfor %}
    }
    impl Gl {
    {% endif %}

    {% for command in feature_set.commands %}
    {{ template_utils.protect(command) }} #[inline] pub unsafe fn {{ command.name|no_prefix }}({{ '&self, ' if options.mx }}{{ command|params }}) -> {{ command.proto.ret|type }} { mem::transmute::<_, extern "system" fn({{ command|params('types') }}) -> {{ command.proto.ret|type }}>({{ 'self._' if options.mx else 'storage::' }}{{ command.name|no_prefix }}.ptr)({{ command|params('names') }}) }
    {% endfor %}

    {{ '}' if options.mx }}
```

An idea of making it a bit more readble, clearly visible which block is the `if options.mx` block and one less template if

Reply:
Indents will be wrong this way, but shouldn't be an issue. definitely much more readable